java/core/lance-jni/src/blocking_scanner.rs

```diff
@@ -143,7 +143,7 @@ fn inner_create_scanner<'local>(
 
     let substrait_opt = env.get_bytes_opt(&substrait_filter_obj)?;
     if let Some(substrait) = substrait_opt {
-        RT.block_on(async { scanner.filter_substrait(substrait).await })?;
+        RT.block_on(async { scanner.filter_substrait(substrait) })?;
```

Comment:
🤔 Why shouldn't we await that future?

Reply:
The method is no longer async.  Now we don't actually compile the filters until scan time (since the input schema may depend on other calls to the scanner builder).  The `scanner.filter` and `scanner.filter_substrait` methods just record whatever the user passes in.

Reply:
If it's no longer async, you could also consider removing the wrapping `RT.block_on`.

Reply:
🤦